analyzers/nuget/analyze.go

```diff
@@ -0,0 +1,130 @@
+package nuget
+
+import (
+	"path/filepath"
+
+	"github.com/fossas/fossa-cli/buildtools/dotnet"
+	"github.com/fossas/fossa-cli/files"
+	"github.com/fossas/fossa-cli/log"
+	"github.com/fossas/fossa-cli/module"
+	"github.com/fossas/fossa-cli/pkg"
+)
+
+func (a *Analyzer) Analyze(m module.Module) (module.Module, error) {
+	log.Logger.Debugf("%#v", m)
+	// Parse lockfile.
+	lockfile, err := dotnet.ReadLockfile(filepath.Join(Dir(m), "obj", "project.assets.json"))
+	if err != nil {
+		return m, err
+	}
+
+	// Compute project graph.
+	projects := make(map[string]dotnet.Manifest)
+	err = Projects(projects, m.BuildTarget)
+	if err != nil {
+		return m, err
+	}
+	root := projects[m.BuildTarget]
+	id := pkg.ID{
+		Type:     pkg.NuGet,
+		Name:     root.Name(),
+		Revision: root.Version(),
+		Location: m.BuildTarget,
+	}
+
+	// Compute package graph.
+	deps := make(map[pkg.ID]pkg.Package)
+	Packages(projects, lockfile, deps, m.BuildTarget)
+
+	m.Imports = deps[id].Imports
+	delete(deps, id)
+	m.Deps = deps
+
+	return m, nil
+}
+
+func Projects(projects map[string]dotnet.Manifest, projectFile string) error {
+	// Break out of cycles.
+	if _, ok := projects[projectFile]; ok {
+		return nil
+	}
+
+	// Read manifest.
+	var manifest dotnet.Manifest
+	err := files.ReadXML(&manifest, projectFile)
+	if err != nil {
+		return err
+	}
+	projects[projectFile] = manifest
+
+	// Get all project references and recurse.
+	for _, reference := range manifest.Projects() {
+		err = Projects(projects, filepath.Join(filepath.Dir(projectFile), dotnet.Path(reference.Include)))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func Packages(projects map[string]dotnet.Manifest, lockfile dotnet.Lockfile, deps map[pkg.ID]pkg.Package, dep string) pkg.ID {
+	log.Logger.Debugf("%#v", dep)
+	if project, ok := projects[dotnet.Path(dep)]; ok {
+		log.Logger.Debugf("%#v", project)
+		name := project.Name()
+		version := project.Version()
+
+		id := pkg.ID{
+			Type:     pkg.NuGet,
+			Name:     name,
+			Revision: version,
+			Location: dep,
+		}
+
+		var imports []pkg.Import
+		for _, ref := range project.Packages() {
+			imports = append(imports, pkg.Import{
+				Target:   ref.Include + "@" + ref.Version,
+				Resolved: Packages(projects, lockfile, deps, ref.Include),
+			})
+		}
+		for _, ref := range project.Projects() {
+			imports = append(imports, pkg.Import{
+				Target:   ref.Include + "@" + ref.Version,
+				Resolved: Packages(projects, lockfile, deps, filepath.Join(filepath.Dir(dep), dotnet.Path(ref.Include))),
+			})
+		}
+
+		deps[id] = pkg.Package{
+			ID:      id,
+			Imports: imports,
+		}
+
+		return id
+	} else {
```

Comment:
if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary)